Refresh the pull request template wording (#2)
zephyr-ai/.github: .github/PULL_REQUEST_TEMPLATE.md

<!--
Hi there. Thanks for submitting a pull request!

> Please include a description of your submitted change in this PR description.

Reviewer: Not a hard suggestion, but a possible alternative:

```suggestion
Please describe your changes in this PR description.
```

> Please describe a description of your submitted change in this PR description.

Author: Ope I messed up this mismash of a sentence.

Reviewer: I like
> Describe a description

> You can also include a link to an active Jira ticket that provides relevant context if there is one.

Reviewer 2: Maybe we should have an example text that the person could use for this?
That way to add a ticket becomes just changing `ABC-123`
```
You can also include a link to an active Jira ticket that provides relevant context if there is one, like this:

[ABC-123](https://zephyrai.atlassian.net/browse/ABC-123)
```

Reviewer 2: Nevermind I see it below.

Author: I actually initially had that! But my thinking was that it's more work and also more error prone to make someone update their ticket number in _two_ places rather than one.

Author: Also, I did keep a sample Jira hyperlink in lower down in the file.

https://policies.zephyrai.bio/sdlc.html
-->

> # Proposed Changes

Reviewer 3: # Proposed Changes 

Seems really big! 

Maybe have it as 

## Proposed Changes 

or

### Proposed Changes

Author: ##### Proposed Changes

Author: I don't really have an opinion on this, but please feel free to follow-up PR if you'd like to reduce it, I'd stamp.

<!--
**Jira ticket**: https://zephyrai.atlassian.net/browse/TICKET
-->

> <!-- Don't forget to motivate your pull request! -->

Reviewer 3: what does motivate mean?

Author: comes from [motivating an argument or thesis](https://www.brandeis.edu/writing-program/resources/students/academic/motivating-the-argument.html). basically, explain why your PR is important, relevant, necessary. convince the reader that it's worth reviewing and worth merging.

Reviewer 3: Ha, Ive rarely come across motivate used in that context. Figure its similar to justify.

Author: Could swap to `describe`, I'm pretty indifferent.
